### PR TITLE
Add  ompsnp_npp , omi_aura and ompstc8_npp YAMLs 

### PR DESCRIPTION
**Pull request by azadeh-gh**
Add Ozone test YAMLs (ompsnp_npp , omi_aura and ompstc8_npp) to parm/atm/obs/testing in GDASApp to reproduce the GSI (HofX, QC and final obs error). https://github.com/NOAA-EMC/JEDI-T2O/issues/57

### CODE REVIEW

**Comment by CoryMartin-NOAA** [comment]
@azadeh-gh does this replace the previous PR you had or compliments it?

**Comment by RussTreadon-NOAA** [comment]
Comparison of PR #386 `feature/sprint-ioda-converters_Azadeh` and PR #438 `OMPS_Validation_Azadeh` show `omi_aura.yaml` and `ompstc8_npp.yaml` to be identical between the two PRs.   `ompsnp_npp.yaml` differ in that PR #386 uses `ozoneTotal` whereas #438 uses `ozoneLayer`.   This is exactly what the PR #438 description states.   This change is [recommended](https://github.com/NOAA-EMC/GDASApp/pull/386#discussion_r1165592052) in PR #386.

My guess is that this PR, #438, replaces #386.  If true, we can close PR #386.

**Comment by azadeh-gh** [comment]
@RussTreadon-NOAA @CoryMartin-NOAA Correct, PR https://github.com/NOAA-EMC/GDASApp/pull/438 replaces PR https://github.com/NOAA-EMC/GDASApp/pull/386

**Review by RussTreadon-NOAA — COMMENTED** [review]
Changes look good.   Two questions:
- Have tests documented in [JEDI-T2O issue #57](https://github.com/NOAA-EMC/JEDI-T2O/issues/57) been rerun with the updated `ompsnp_npp.yaml` in this PR?
- Will the VertInterp code sprint require the three ozone yamls in this PR to be updated?

File: parm/atm/obs/testing/omi_aura.yaml [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+ # Gross check
+- filter: Background Check
+  filter variables:
+  - name: ozoneTotal
+  threshold: 5.0
+  action:
+    name: reject
+
+# Gross error check with (O - B) / ObsError greater than threshold.
```
Comment (HaixiaLiu-NOAA) [inline parm/atm/obs/testing/omi_aura.yaml]: are these (O-B)/ObsError checks newly added in JEDI? I did not see these QCs in the GSI. Similar to this, in the ompstc8_npp.yaml. the (O-B)/ObsError checks are added which I do not think exist in GSI. The changes look good to me if they are done on purpose.

Comment (azadeh-gh) [reply parm/atm/obs/testing/omi_aura.yaml]: @RussTreadon-NOAA Yes I updated the the plots related to omps_npp in https://github.com/NOAA-EMC/JEDI-T2O/issues/57

**Comment by CoryMartin-NOAA** [comment]
@azadeh-gh can you update the PR with a more descriptive name of what is being changed/added here? When it gets merged in it will be more clear for users/developers that way.

File: parm/atm/obs/testing/omi_aura.yaml (hunk 1 above)
Comment (CoryMartin-NOAA) [reply parm/atm/obs/testing/omi_aura.yaml]: @HaixiaLiu-NOAA is that not what https://github.com/NOAA-EMC/GSI/blob/14ae595af1b03471287d322596d35c0665336e95/src/gsi/setupoz.f90#L1363-L1380 is doing?

**Comment by CoryMartin-NOAA** [comment]
> * Will the VertInterp code sprint require the three ozone yamls in this PR to be updated?

Not for this sprint next week, but Jerome and I hope to combine AtmVertInterpLay into the ColumnRetrieval operator so that we can reduce the total # of operators, since it should be able to be generalized enough to handle these ozone observations

**Comment by azadeh-gh** [comment]
@CoryMartin-NOAA Updated the PR name and comment.

File: parm/atm/obs/testing/omi_aura.yaml (hunk 1 above)
Comment (HaixiaLiu-NOAA) [reply parm/atm/obs/testing/omi_aura.yaml]: @CoryMartin-NOAA That is for OMPS limb profiler (OMPSLP) data, not for omi_aura or ompstc8_npp.

**Comment by RussTreadon-NOAA** [comment]
> > * Will the VertInterp code sprint require the three ozone yamls in this PR to be updated?
> 
> Not for this sprint next week, but Jerome and I hope to combine AtmVertInterpLay into the ColumnRetrieval operator so that we can reduce the total # of operators, since it should be able to be generalized enough to handle these ozone observations

Thank you @CoryMartin-NOAA for sharing the long term plan.

File: parm/atm/obs/testing/omi_aura.yaml (hunk 1 above)
Comment (RussTreadon-NOAA) [reply parm/atm/obs/testing/omi_aura.yaml]: > @RussTreadon-NOAA Yes I updated the the plots related to omps_npp in [NOAA-EMC/JEDI-T2O#57](https://github.com/NOAA-EMC/JEDI-T2O/issues/57)

Great!  Thank you @azadeh-gh  for the confirmation.

**Review by RussTreadon-NOAA — DISMISSED** [review]
My questions have been answered.   Plots in JEDI-T2O issue [#57](https://github.com/NOAA-EMC/JEDI-T2O/issues/57) indicate that yamls work.  No more questions from me.

Approve.

File: parm/atm/obs/testing/omi_aura.yaml (hunk 1 above)
Comment (CoryMartin-NOAA) [reply parm/atm/obs/testing/omi_aura.yaml]: @HaixiaLiu-NOAA ah okay, sorry I didn't realize it only applied to certain observation types, @azadeh-gh can you confirm this and make the changes as necessary for the O-B check?

**Comment by azadeh-gh** [comment]
@HaixiaLiu-NOAA the flowchart of the QC and the plots are https://github.com/NOAA-EMC/JEDI-T2O/issues/57
I added O-B Obserror check to omi_aura.yaml based on [setupoz](https://github.com/NOAA-EMC/GSI/blob/14ae595af1b03471287d322596d35c0665336e95/src/gsi/setupoz.f90#L1363-L1380) similar to ompsnp_npp.yaml. The correlation between h(x) vs obs passed QC increases after adding O-B checks (from R2= 9153 to R2=9685).

ompstc8_npp.yaml does not have (O-B)/ObsError checks.

**Review by HaixiaLiu-NOAA — APPROVED** [review]
all look good to me. Thanks.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Review by RussTreadon-NOAA — APPROVED** [review]
Approve.

**Pull request merged**